collect_app/src/main/java/org/odk/collect/android/forms/DatabaseMediaFileRepository.java

```diff
@@ -12,7 +12,10 @@
     @Override
     public List<File> getAll(String jrFormID, String formVersion) {
         String formMediaPath = new FormsDao().getFormMediaPath(jrFormID, formVersion);
-        File[] files = new File(formMediaPath).listFiles();
+        File[] files = new File[0];
+        if (formMediaPath != null) {
```

Comment:
I think we need a test to drive this out so we avoid regressing at any point. As far as I can see the situation that gives us the NPE (for the previous code) was if `getAll` was queried for a `jrFormID` and `formVersion` not in the database. If that's right we could have a `DatabaseMediaFileRepositoryTest` similar to the `DatabaseFormRepositoryTest` and a test that just checks we get back an empty list when there is nothing in the database.

Reply:
Done.